pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java

```diff
@@ -92,26 +99,31 @@ public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
       } else {
         tableMaxNumTasks = Integer.MAX_VALUE;
       }
-      List<SegmentZKMetadata> offlineSegmentsZKMetadata = _clusterInfoAccessor.getSegmentsZKMetadata(tableName);
+      List<SegmentZKMetadata> segmentsZKMetadata;
```

Comment:
We can read all the segments ZK metadata here, and for real-time table, remove the segments that is not completed

Reply:
It was right. I've removed wat was on function to get segment right on on the if section